core/src/services/s3/core.rs

```diff
@@ -257,14 +257,32 @@ impl S3Core {
     ) -> http::request::Builder {
         if let Some(checksum_algorithm) = self.checksum_algorithm.as_ref() {
             let checksum = match checksum_algorithm {
-                S3ChecksumAlgorithm::Crc32c => {
-                    BASE64_STANDARD.encode(crc32c::crc32c(body.to_vec().as_slice()).to_be_bytes())
+                ChecksumAlgorithm::Crc32c => {
+                    let mut crc = 0u32;
+                    body.clone()
```

Comment:
This clone should be okay, since it does not do any allocation (?)

Reply:
Yep, this clone is `Arc::clone`.